crates/ruma-common/src/authentication.rs

```diff
@@ -3,11 +3,22 @@
 use ruma_serde::StringEnum;
 
 /// Access token types.
+///
+/// This type can hold an arbitrary string. To check for formats that are not available as a
+/// documented variant here, use its string representation, obtained through `.as_str()`.
 #[derive(Clone, Debug, PartialEq, Eq, StringEnum)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
```

Comment:
I didn't apply non_exhaustive here before because there's an undocumented variant anyways that's not supposed to be part of the public API. I guess it's better to further enforce that though.

In this case (`_Custom` variant) I think there is definitely no point in being able to turn off `non_exhaustive` so we might as well have that unconditionally.